packages/accounts-controller/src/AccountsController.ts

```diff
@@ -248,12 +292,82 @@ export class AccountsController extends BaseController<
   }
 
   /**
-   * Returns the selected internal account.
+   * Returns the last selected evm account.
    *
    * @returns The selected internal account.
    */
   getSelectedAccount(): InternalAccount {
-    return this.getAccountExpect(this.state.internalAccounts.selectedAccount);
+    const selectedAccount = this.getAccountExpect(
+      this.state.internalAccounts.selectedAccount,
+    );
+    if (isEvmAccountType(selectedAccount.type)) {
+      return selectedAccount;
+    }
+
+    const accounts = this.listAccounts();
+
+    if (!accounts.length) {
+      // !Should never reach this.
+      throw new Error('AccountsController: No EVM accounts');
+    }
+
+    let lastSelectedEvmAccount = accounts[0];
+    lastSelectedEvmAccount = accounts.reduce((prevAccount, currentAccount) => {
+      if (
+        // When the account is added, lastSelected will be set
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+        currentAccount.metadata.lastSelected! >
+        // When the account is added, lastSelected will be set
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+        prevAccount.metadata.lastSelected!
+      ) {
+        return currentAccount;
+      }
+      return prevAccount;
+    }, lastSelectedEvmAccount); // Safe indexing, since we checked for .length already
+
+    return lastSelectedEvmAccount;
```

Comment:
What about just having 1 `return` here? The `let` seems a bit unnecessary?

Reply:
we can use const, but the initial value will have to be accounts[0]